packages/cli/src/databases/repositories/execution.repository.ts

```diff
@@ -71,6 +75,14 @@ export class ExecutionRepository extends Repository<ExecutionEntity> {
 		private readonly executionDataRepository: ExecutionDataRepository,
 	) {
 		super(ExecutionEntity, dataSource.manager);
+
+		if (!inTest) {
+			if (config.getEnv('executions.pruneData')) {
```

Comment:
We could disable pruning in the relevant test setups, or in the global test setup, instead of adding a test check here.